lib/optimizely.rb

```diff
@@ -92,7 +95,13 @@ def initialize(
                           StaticProjectConfigManager.new(datafile, @logger, @error_handler, skip_json_validation)
                         end
       @decision_service = DecisionService.new(@logger, @user_profile_service)
-      @event_builder = EventBuilder.new(@logger)
+
+      @event_processor = if event_processor.respond_to?(:process)
```

Comment:
what does it mean `respond_to` ?

Reply:
Checks instance method `process` exist, similarly we are checking for config_manager; `config_manager.respond_to?(:get_config)`